runtime/compiler/runtime/MetaData.cpp

```diff
@@ -1138,7 +1147,11 @@ populateBodyInfo(
       }
    else
       {
-      if (vm->isAOT_DEPRECATED_DO_NOT_USE())
+      if (vm->isAOT_DEPRECATED_DO_NOT_USE()
+#if defined(JITSERVER_SUPPORT)
+         || comp->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER
```

Comment:
Since we have the comp object, a better test could be comp->isOutProcessCompilation()